src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -49,6 +49,8 @@ public Http3Connection(Http3ConnectionContext context)
             _timeoutControl = new TimeoutControl(this);
             _context.TimeoutControl ??= _timeoutControl;
 
+            _errorCodeFeature = context.ConnectionFeatures.Get<IProtocolErrorCodeFeature>()!;
```

Comment:
I hated this approach when I added it. Maybe better to expose an exception type in the transport that has an error code on it? Then when we call abort, the error code will be set.

Reply:
The exception would have to be in connection abstractions so Kestrel can throw it and the QUIC transport can read it.

Looking at where `IProtocolErrorCodeFeature` is used, `ConnectionAbortedException` would be a good place to add a `long? ProtocolError { get; }` property. How come a feature was used instead?

@davidfowl

Reply:
I also prefer the protocol error being an argument to abort rather than a property that's set beforehand. What if we made a `Http3ConnectionAbortedException` with a `ProtocolError` property and do an `as` check in `Http3Connection.Abort(ConnectionAbortedException)`? I think I prefer that slightly to just adding the property directly to `ConnectionAbortedException`.

Reply:
Because I didn't want to modify ConnectionAbortedException I believe at the time.